orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationAiModel.java

```diff
@@ -114,4 +117,58 @@ public class OrchestrationAiModel {
   LLMModuleConfig createConfig() {
     return LLMModuleConfig.create().modelName(name).modelParams(params).modelVersion(version);
   }
+
+  /**
+   * Additional parameter on this model.
+   *
+   * @param key the parameter key.
+   * @param value the parameter value, nullable.
+   * @return A new model with the additional parameter.
+   */
+  @Nonnull
+  public OrchestrationAiModel withParam(@Nonnull final String key, @Nullable final Object value) {
+    final var params = new LinkedHashMap<>(getParams());
+    params.put(key, value);
+    return withParams(params);
+  }
+
+  /**
+   * Additional parameter on this model.
+   *
+   * @param param the parameter key.
+   * @param value the parameter value, nullable.
+   * @return A new model with the additional parameter.
+   */
+  @Nonnull
+  public OrchestrationAiModel withParam(
+      @Nonnull final Parameter param, @Nullable final Object value) {
+    return withParam(param.value, value);
+  }
+
+  /** Parameter key for a model. */
+  @RequiredArgsConstructor
+  public enum Parameter {
+    /** The maximum number of tokens to generate. */
+    MAX_TOKENS("max_tokens"),
```

Comment:
Do you think it would be feasible to also include the datatype class for these properties?

e.g. something like:
```suggestion
    MAX_TOKENS("max_tokens", Integer.class),
```

and use a generic in the `withParam` to type-check the value being passed on the API level? Or maybe this is overkill for this API?

Reply:
I like the idea, changing the solution was easy enough.